src/librustc_privacy/lib.rs

```diff
@@ -433,15 +442,6 @@ impl<'b, 'a, 'tcx> TypeVisitor<'tcx> for ReachEverythingInTheInterfaceVisitor<'b
 
         ty.super_visit_with(self)
     }
-
-    fn visit_trait_ref(&mut self, trait_ref: ty::TraitRef<'tcx>) -> bool {
-        if let Some(node_id) = self.ev.tcx.hir.as_local_node_id(trait_ref.def_id) {
-            let item = self.ev.tcx.hir.expect_item(node_id);
```

Comment:
Can try, but I think the impl that's *more* messed up is the other `SearchInterfaceForPrivateItemsVisitor` (the test failures I'm looking at, though there are others I haven't checked yet).

Here, if we visit item_def_id instead, won't we skip the trait_ref completely? Or does visiting the item also descend into trait_ref. I'm afraid we're somehow only going to go to the `substs` right away.